### PR TITLE
build: verify that the protobufs in the spec actually compile

### PR DESCRIPTION
**Pull request by jdef**
* TravisCI integration that verifies the protobufs in the spec actually compile
* add build badge to README

### CODE REVIEW

**Comment by jdef** [comment]
blocked by #13

**Comment by jdef** [comment]
rebase to master

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+language: go
+go:
```
Comment (jieyu) [inline .travis.yml]: nit: Let's have an empty line between sections.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+before_install:
+  - curl -LO https://github.com/google/protobuf/releases/download/v3.3.0/protoc-3.3.0-linux-x86_64.zip
+  - unzip protoc-3.3.0-linux-x86_64.zip
+  - chmod +x bin/protoc
+install:
+  - go get -u github.com/golang/protobuf/proto
+  - go get -u github.com/golang/protobuf/protoc-gen-go
+  - go get -u google.golang.org/grpc
+  - cat spec.md | sed -n -e '/```protobuf$/,/```$/ p' | sed -e '1i syntax = "proto3";\npackage spec;' -e 's@^```.*$@////////@g' > spec.proto
+script: bin/protoc --go_out=plugins=grpc:. *.proto
```
Comment (jieyu) [inline .travis.yml]: nit: put the command in the following line

**Comment by saad-ali** [comment]
LGTM
Please squash the commits and we can merge

**Pull request merged**